framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -870,16 +870,24 @@ private AnnotatedDeclaredType(
             wasRaw = !declty.getTypeArguments().isEmpty() && type.getTypeArguments().isEmpty();
 
             TypeMirror encl = type.getEnclosingType();
-            if (encl.getKind() == TypeKind.DECLARED) {
-                this.enclosingType =
-                        (AnnotatedDeclaredType) createType(encl, atypeFactory, declaration);
-            } else if (encl.getKind() != TypeKind.NONE) {
-                throw new BugInCF(
-                        "AnnotatedDeclaredType: unsupported enclosing type: "
-                                + type.getEnclosingType()
-                                + " ("
-                                + encl.getKind()
-                                + ")");
+            switch (encl.getKind()) {
+                case NONE:
+                    this.enclosingType = null;
+                    break;
+                case DECLARED:
+                    this.enclosingType =
+                            (AnnotatedDeclaredType) createType(encl, atypeFactory, declaration);
+                    break;
+                case ERROR:
+                    throw new BugInCF(
```

Comment:
Maybe I'm missing something, but I don't see how this is different from the default case.

Reply:
An error is not unexpected, but indicates a problem earlier (such as code that is not compilable).

Reply:
OK, I must have been running into that bug.